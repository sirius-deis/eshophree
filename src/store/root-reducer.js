import { combineReducers } from "redux";

import categoryReducer from "./category/category.reducer";
import userReducer from "./user/user.reducer";
import cartReducer from "./cart/cart.reducer";

const rootReducer = combineReducers({
    category: categoryReducer,
    user: userReducer,
    cart: cartReducer,
});

export default rootReducer;
