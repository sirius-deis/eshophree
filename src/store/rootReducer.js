import { combineReducers } from 'redux';

import userReducer from './user/user.reducer';
import categoryReducer from './category/category.reducer';

const rootReducer = combineReducers({
  user: userReducer,
  category: categoryReducer,
});

export default rootReducer;
