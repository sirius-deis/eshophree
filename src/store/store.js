import { createStore, applyMiddleware } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import rootReducer from './rootReducer';

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const middlewares = [thunk];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);
}

export const store = createStore(persistedReducer, applyMiddleware(...middlewares));

export const persistor = persistStore(store);
