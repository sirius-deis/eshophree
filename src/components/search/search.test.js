import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Search from './search';

describe('Search component', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Search />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should fire click event and show dropdown', () => {
    render(
      <Provider store={store}>
        <Search />
      </Provider>,
    );
    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
    fireEvent.click(screen.queryAllByRole('button')[0]);
    expect(screen.getByRole('listbox')).toBeInTheDocument();
  });
  it('should fire mouse events and show dropdown', () => {
    render(
      <Provider store={store}>
        <Search />
      </Provider>,
    );
    fireEvent.mouseEnter(screen.queryAllByRole('button')[0]);
    expect(screen.getByRole('listbox')).toBeInTheDocument();
    fireEvent.mouseLeave(screen.queryAllByRole('button')[0]);
    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
  });
});
