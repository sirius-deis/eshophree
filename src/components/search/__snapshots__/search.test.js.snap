// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match snapshot 1`] = `
<div>
  <form
    class="sc-bgqQcB iwyGvT"
  >
    <div
      class="sc-kPTPQs kMxmqA"
    >
      <button
        class="sc-fFGjHI sc-iVCKna kVWsUU flGjLA"
        style="background-color: transparent;"
        type="button"
      >
        <div
          class="sc-ewnqHT fkubsD"
        >
          All categories
        </div>
        <svg
          fill="currentColor"
          font-size="22"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
      </button>
    </div>
    <input
      class="sc-gTRrQi gcuGNn"
      type="text"
    />
    <button
      class="sc-fFGjHI sc-dicizt kVWsUU hluNsG"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </button>
  </form>
</div>
`;
